### DISCUSSION TITLE
Filter on same @collection twice

### DISCUSSION THREAD

**Discussion by odama626** [pocketbase/pocketbase#2941] [Q&A] [answered]
Imagine the following schema
![image](https://github.com/pocketbase/pocketbase/assets/11347689/616e6f73-2214-4bd0-9af4-7a0390b2c7ee)

I am trying to create an api rule for the users list that allows a user to see all users that they share a project with and I am running into a problem because I think I need to be able to join `@collections.users_projects` twice

this is similar to what I am trying now
```
id ?= @collections.users_projects.user 
&& @collections.users_projects.project = @collections.users_projects.project
&& @collections.users_projects.user = @request.auth.id
```

I think for it to actually work I would need to be able to do something like this?

```
id ?= @collections.users_projects.user 
&& @collections.users_projects.project = @collections.users_projects2.project 
&& @collections.users_projects2.user = @request.auth.id
```

or maybe if we could alias the `@collections.*` references like joins?

**Comment by ganigeorgiev** [comment, marked as answer]
There is no support for `@collection.*` aliases at the moment (_I've added it in my todo and may consider it in the future together with the indirect relation lookups_).

As a workaround, you can create a helper View collection with your custom SQL query or [register a custom Go route](https://pocketbase.io/docs/router/#register-new-route) (_with the upcoming v0.17.0 there will be also an option to do that with JS_).

**Reply by odama626** [reply]
using a view collection and then pointing to it with the api rules worked fantastically, thanks!

I must say, pocketbase is a real delight to use
